manifests/compute/foreman_compute.pp

```diff
@@ -0,0 +1,13 @@
+# Provides a common class to install foreman-compute
+# used by multiple compute resources
+#
+# === Parameters:
+#
+# $version::  Package version to install, defaults to installed
+#
+class foreman::compute::foreman_compute ( $version = 'installed' ) {
```

Comment:
Would you mind marking this class as private, so people have to go through the main classes?

Just call private() and increase the stdlib dependency to >= 4.4.0 in metadata.json.